### PR TITLE
Resolve some linter/compiler warning messages

### PR DESCRIPTION
**Pull request by tcw165**

### CODE REVIEW

File: src/tests/copy_count_tracker.hpp [hunk 1]
```diff
@@ -54,8 +54,11 @@ class copy_count_tracker
     {
         return rpp::source::create<copy_count_tracker>([this, count](const auto& sub)
         {
-            for (size_t i = 0; i < count; ++i)
+            for (size_t i = 0; i < count; ++i) {
+                if (!sub.is_subscribed()) { break; }
+
                 sub.on_next(*this);
+            }
             sub.on_completed();
```
Comment (victimsnino) [inline src/tests/copy_count_tracker.hpp:62]: Code-style =)
```suggestion
            for (size_t i = 0; i < count; ++i) 
            {
                if (!sub.is_subscribed())
                    break;

                sub.on_next(*this);
            }
            sub.on_completed();
```

File: src/tests/mock_observer.hpp [hunk 2]
```diff
@@ -20,7 +20,8 @@ template<typename Type>
 class mock_observer : public rpp::interface_observer<Type>
 {
 public:
-    mock_observer(bool copy_values = true) : m_state{std::make_shared<State>(copy_values)} {}
+    explicit mock_observer(bool copy_values = true) : m_state{std::make_shared<State>(copy_values)} {}
+    ~mock_observer() noexcept = default;
```
Comment (victimsnino) [inline src/tests/mock_observer.hpp:24]: Do we really need it? Anyway, it must have `override`

File: src/tests/copy_count_tracker.hpp (hunk 1 above)
Comment (tcw165) [reply src/tests/copy_count_tracker.hpp:62]: Sounds good :)

File: src/tests/mock_observer.hpp (hunk 2 above)
Comment (tcw165) [reply src/tests/mock_observer.hpp:24]: No, we don't really need the dtor.

File: src/tests/copy_count_tracker.hpp (hunk 1 above)
Comment (victimsnino) [reply src/tests/copy_count_tracker.hpp:62]: Waiting for commit then

Comment (tcw165) [reply src/tests/copy_count_tracker.hpp:62]: Addressed in the latest commit.

File: src/tests/mock_observer.hpp (hunk 2 above)
Comment (tcw165) [reply src/tests/mock_observer.hpp]: Addressed in the latest commit.

File: src/tests/copy_count_tracker.hpp [hunk 3]
```diff
@@ -54,8 +54,11 @@ class copy_count_tracker
     {
         return rpp::source::create<copy_count_tracker>([this, count](const auto& sub)
         {
-            for (size_t i = 0; i < count; ++i)
+            for (size_t i = 0; i < count; ++i) {
+                if (!sub.is_subscribed()) { break; }
+
                 sub.on_next(*this);
+            }
```
Comment (victimsnino) [inline src/tests/copy_count_tracker.hpp:61]: Same as below
```suggestion
            for (size_t i = 0; i < count && sub.is_subscribed(); ++i) 
                sub.on_next(*this);
```

Comment (tcw165) [reply src/tests/copy_count_tracker.hpp]: Addressed in the latest commit (overriden)

**Review by victimsnino — APPROVED** [review]

**Pull request merged**